canvas.go

```diff
@@ -48,8 +52,32 @@ type line struct {
 	p0, p1 image.Point
 }
 
+var (
+	pathSanitizer = regexp.MustCompile(`\W+`)
+	piSanitizer   = regexp.MustCompile(`<\?xml[^>]*>`)
```

Comment:
Strictly speaking "<?xml..." isn't a processing instruction, you should probably be checking for "<?..." I think?